src/bert_featurizer.py

```diff
@@ -0,0 +1,29 @@
+import pandas as pd
+import torch
+from transformers import AutoModel, AutoTokenizer
+
+
+class BertFeatureExtractor:
+    """Extract [CLS] embedding feature from any untrained bert-like models"""
+
+    def __init__(self, model_name: str):
+        """
+        :param model_name: name of model from hugging face
+        """
+        self.model_name = model_name
+
+        self.model = AutoModel.from_pretrained(self.model_name)
+        self.tokenizer = AutoTokenizer.from_pretrained(self.model_name)
+
+    @torch.no_grad()
+    def extract_features(self, row: pd.Series):
```

Comment:
I believe it's better to make this function run on the series of tests, not just one text. Just for computational efficiency: parallel inference is faster than sequential inference.

So, method `extract_features` should accept one argument `texts` of type `pd.Series`, where index comes from the index of text in the original train dataframe, and values of the series are the texts (one element in series - one text)

Reply:
Also, please add docstring and annotation for output value, aka   
`def extract_features(...) -> pd.DataFrame:`

Reply:
It seems Shamil has fixed all issues, mark conversation as resolved.